base/setup/lib/utils/osdetect.c

```diff
@@ -755,90 +749,59 @@ FindNTOSInstallations(
-    PPARTENTRY PartEntry;
+    PLIST_ENTRY Entry;
+    PVOLENTRY Volume;
+    BOOLEAN CheckVolume;
 
     List = CreateGenericList();
-    if (List == NULL)
+    if (!List)
         return NULL;
 
-    /* Loop each available disk ... */
-    Entry = PartList->DiskListHead.Flink;
-    while (Entry != &PartList->DiskListHead)
+    /* Loop each available volume */
+    for (Entry = PartList->VolumesList.Flink;
+         Entry != &PartList->VolumesList;
+         Entry = Entry->Flink)
     {
-        DiskEntry = CONTAINING_RECORD(Entry, DISKENTRY, ListEntry);
-        Entry = Entry->Flink;
-
-        DPRINT("Disk #%d\n", DiskEntry->DiskNumber);
-
-        /* ... and for each disk, loop each available partition */
-
-        /* First, the primary partitions */
-        Entry2 = DiskEntry->PrimaryPartListHead.Flink;
-        while (Entry2 != &DiskEntry->PrimaryPartListHead)
-        {
-            PartEntry = CONTAINING_RECORD(Entry2, PARTENTRY, ListEntry);
-            Entry2 = Entry2->Flink;
-
-            ASSERT(PartEntry->DiskEntry == DiskEntry);
-
-            DPRINT("   Primary Partition #%d, index %d - Type 0x%02x, IsLogical = %s, IsPartitioned = %s, IsNew = %s, FormatState = %lu -- Should I check it? %s\n",
-                   PartEntry->PartitionNumber, PartEntry->PartitionIndex,
-                   PartEntry->PartitionType, PartEntry->LogicalPartition ? "TRUE" : "FALSE",
-                   PartEntry->IsPartitioned ? "TRUE" : "FALSE",
-                   PartEntry->New ? "Yes" : "No",
-                   PartEntry->FormatState,
-                   ShouldICheckThisPartition(PartEntry) ? "YES!" : "NO!");
-
-            if (ShouldICheckThisPartition(PartEntry))
-                FindNTOSInstallations(List, PartList, PartEntry);
-        }
-
-        /* Then, the logical partitions (present in the extended partition) */
-        Entry2 = DiskEntry->LogicalPartListHead.Flink;
-        while (Entry2 != &DiskEntry->LogicalPartListHead)
-        {
-            PartEntry = CONTAINING_RECORD(Entry2, PARTENTRY, ListEntry);
-            Entry2 = Entry2->Flink;
-
-            ASSERT(PartEntry->DiskEntry == DiskEntry);
-
-            DPRINT("   Logical Partition #%d, index %d - Type 0x%02x, IsLogical = %s, IsPartitioned = %s, IsNew = %s, FormatState = %lu -- Should I check it? %s\n",
-                   PartEntry->PartitionNumber, PartEntry->PartitionIndex,
-                   PartEntry->PartitionType, PartEntry->LogicalPartition ? "TRUE" : "FALSE",
-                   PartEntry->IsPartitioned ? "TRUE" : "FALSE",
-                   PartEntry->New ? "Yes" : "No",
-                   PartEntry->FormatState,
-                   ShouldICheckThisPartition(PartEntry) ? "YES!" : "NO!");
-
-            if (ShouldICheckThisPartition(PartEntry))
-                FindNTOSInstallations(List, PartList, PartEntry);
-        }
+        PPARTENTRY PartEntry;
+        Volume = CONTAINING_RECORD(Entry, VOLENTRY, ListEntry);
+        PartEntry = Volume->PartEntry; // FIXME
```

Comment:
again: in case you know it, try to be as specific as possible about what should be fixed later in the comment.

Reply:
Yep, this one is kind of a "fixme/todo: remove if possible" because it is used only for doing the dprint that is below.